src/com/lilithsthrone/game/inventory/clothing/AbstractClothingType.java

```diff
@@ -1393,51 +1402,81 @@ public ClothingSet getClothingSet() {
 //		return SVGStringEquippedMap;
 //	}
 	
-	private void addSVGStringMapping(Colour colour, Colour colourSecondary, Colour colourTertiary, String s) {
+	private void addSVGStringMapping(Colour colour, Colour colourSecondary, Colour colourTertiary, String pattern, String s) {
+		if(pattern == null) {
+			pattern = "none"; // The map does not contain null as a key.
+		}
 		if(SVGStringMap.get(colour)==null) {
```

Comment:
putIfAbsent can replace this entire tree; if you're not willing to look into the details, remind me later when this is merged and I'll look into it